pkg/genericapiserver/genericapiserver.go

```diff
@@ -269,6 +269,10 @@ type GenericAPIServer struct {
 	enableOpenAPISupport   bool
 	openAPIInfo            spec.Info
 	openAPIDefaultResponse spec.Response
+
+	// PostStartHooks is called in a go func after the server go func is launched, but not necessarily after
+	// the server is listening.  It may kill the process with a panic if it wishes to.
```

Comment:
> Could return an error indicating something fatal, or just make the comment more explicit. Maybe something like:

Updated.   Also updated to start the posthooks after the server is listening because that's just easier.